datafusion/core/src/execution/context.rs

```diff
@@ -477,6 +429,88 @@ impl SessionContext {
         }
     }
 
+    async fn create_custom_table(
+        &self,
+        cmd: &CreateExternalTable,
+    ) -> Result<Arc<DataFrame>> {
+        let factory = &self.table_factories.get(&cmd.file_type).ok_or_else(|| {
+            DataFusionError::Internal(format!(
```

Comment:
I don't think this is an internal error -- it can happen when the user provides a bad query, right? How about `DataFusionError::Execution` instead?

Reply:
You're right - I got sloppy, sorry :(